src/main/flight/imu.c

```diff
@@ -196,6 +198,23 @@ static void imuTransformVectorBodyToEarth(t_fp_vector * v)
     v->V.Z = z;
 }
 
+void imuTransformVectorEarthToBody(t_fp_vector_def * v) {
+    const float x = (q0q0 + q1q1 - q2q2 - q3q3) * v->X + 2*(q1q2 + q0q3) * v->Y + 2*(q1q3 - q0q2) * v->Z;
```

Comment:
you can still use (faster) matrix rotation here ... just transpose rotation matrix to get inverse rotation